ZENKO-949 feature: prometheus chart

giacope opened the pull request:
#259 and #263 in one PR but with more appropriate commits.

This PR in a nutshell:

- Add the Prometheus chart source to this repo.
- Update the Prometheus chart to use StatefulSets instead of Deployments to manage the `server` and `alertmanager` pods.
- Change the default number of replicas to 2 for non-single-node deployments.

giacope committed:
feat(ZENKO-949): update Prometheus chart to use StatefulSets

The Prometheus chart will now be managed through StatefulSets.
This allows the Prometheus "servers" to scale to more than one
replica, and PVCs won't be deleted with the chart release.

giacope committed:
feat(ZENKO-949): add prometheus chart

Pulled from upstream helm/charts: cb9da80

bert-e commented:
# Hello giacomoguiulfo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @ssalaues

* @vrancurel

* @giacomoguiulfo

* @NicolasT

giacope committed:
feat(ZENKO-949): update default replicaCount for Prometheus

There will now be 2 Prometheus servers deployed by default
on non single-node Zenko deployments.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/0.9`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of Release Engineering will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of Release Engineering now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/0.9`




Please check the status of the associated issue ZENKO-949.

Goodbye giacomoguiulfo.
